sgl-kernel/src/sgl-kernel/csrc/rotary_embedding.cu

```diff
@@ -0,0 +1,205 @@
+// Reference: https://github.com/vllm-project/vllm/blob/main/csrc/pos_encoding_kernels.cu
```

Comment:
- Since `batched_rotary_embedding` is unnecessary, we don't need to port it ref https://github.com/sgl-project/sglang/pull/2997
- As discussed with @yzh119, it's recommended to port the implementation from FlashIfner rather than vLLM

Reply:
As discussed offline
1. Remove vllm macro and batched rope
2. [following PR] replace `ops.rotary_embedding` with sgl-kernel's rope and verify performance on CIs
3. [following PR] optimize cuda impl 

1 is done

Reply:
@zhyncs do we need to release a new sgl-kernel version for the rope change?

Reply:
yeah I think so